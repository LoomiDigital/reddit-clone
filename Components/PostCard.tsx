import React, { SyntheticEvent, useEffect, useState } from "react";
import Link from "next/link";

import { loadPage } from "@d20/utils/loadPage";
import { useAddVote } from "@d20/hooks/useAddVote";
import { useComments } from "@d20/hooks/useComments";
import { PostAttributesFragment } from "@d20/generated/graphql";

import TimeAgo from "react-timeago";

import {
  ArrowDownIcon,
  ArrowUpIcon,
  BookmarkIcon,
  ChatBubbleOvalLeftEllipsisIcon,
  EllipsisHorizontalIcon,
  GiftIcon,
  ShareIcon,
} from "@heroicons/react/24/outline";
import Avatar from "./Avatar";

interface Props {
  post: PostAttributesFragment;
}

const PostCard = ({ post }: Props) => {
  const loadSubredditPage = (e: SyntheticEvent) => {
    loadPage(e, `/subreddit/${post.subreddit_topic}`);
  };

  const [hasMounted, setHasMounted] = useState<boolean>(false);
  const { upVote, vote, displayVotes } = useAddVote(post);
  const { comments, loading } = useComments(post);

  useEffect(() => {
    setHasMounted(true);
  }, []);

  return (
    <Link passHref href={`/post/${post.id}`}>
      <div className="flex cursor-pointer rounded-md border border-gray-300 bg-white shadow-sm hover:border hover:border-gray-600">
        <div className="flex flex-col items-center justify-start space-y-1 rounded-l-md bg-gray-50 p-4 text-gray-400">
          <ArrowUpIcon
            onClick={(e) => upVote(e, true)}
            title="upvote"
            className={`voteButtons cursor-pointer hover:text-red-400 ${
              vote && "text-red-400"
            }`}
          />

          <p title="votes" className="votes text-xs font-bold text-black">
            {displayVotes}
          </p>

          <ArrowDownIcon
            onClick={(e) => upVote(e, false)}
            title="downvote"
            className={`voteButtons cursor-pointer hover:text-blue-400 ${
              !vote && "text-blue-400"
            }`}
          />
        </div>
        <div className="p-3 pb-1">
          <div className="flex items-center space-x-2">
            <Avatar seed={post?.subreddit_topic!} />
            <p className="text-xs text-gray-400">
              <span
                onClick={(e) => loadSubredditPage(e)}
                className="font-bold text-black hover:text-blue-400 hover:underline"
              >
                r/{post.subreddit_topic}
              </span>

              <span>
                â€¢ Posted by u/{post.username}{" "}
                {hasMounted ? (
                  <TimeAgo date={post.created_at} autoPlay={false} />
                ) : (
                  "Loading..."
                )}
              </span>
            </p>
          </div>

          <div className="py-4">
            <h2 className="text-xl font-semibold">{post.title}</h2>
            <p className="mt-2 text-sm font-light">{post.body}</p>
          </div>
          {post.image && <img className="w-full" src={post.image} alt="" />}

          <div className="flex space-x-4 text-gray-400 ">
            <div className="postButtons">
              <ChatBubbleOvalLeftEllipsisIcon className="h-6 w-6" />
              <p>{loading ? 0 : comments?.length} Comments</p>
            </div>
            <div className="postButtons">
              <GiftIcon className="h-6 w-6" />
              <p className="hidden sm:inline">Award</p>
            </div>
            <div className="postButtons">
              <ShareIcon className="h-6 w-6" />
              <p className="hidden sm:inline">Share</p>
            </div>
            <div className="postButtons">
              <BookmarkIcon className="h-6 w-6" />
              <p className="hidden sm:inline">Save</p>
            </div>
            <div className="postButtons">
              <EllipsisHorizontalIcon className="h-6 w-6" />
            </div>
          </div>
        </div>
      </div>
    </Link>
  );
};

export default PostCard;
