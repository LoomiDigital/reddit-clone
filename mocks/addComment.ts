import { MockedResponse } from "@apollo/client/testing";
import { AddCommentDocument } from "@d20/generated/graphql";

export const mockAddComment: MockedResponse = {
  request: {
    query: AddCommentDocument,
    variables: {
      post_id: 1,
      text: "This is a new comment",
      username: "Buck",
    },
  },
  result: {
    data: {
      addComment: {
        id: 3,
        text: "This is a new comment",
        post_id: 1,
        created_at: "2023-05-08T18:10:10.831966Z",
        username: "Buck",
      },
    },
  },
};
