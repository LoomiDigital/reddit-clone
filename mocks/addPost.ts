import { MockedResponse } from "@apollo/client/testing";
import { AddPostDocument } from "@d20/generated/graphql";

export const mockAddPost: MockedResponse = {
  request: {
    query: AddPostDocument,
    variables: {
      title: "This is a new post",
      body: "A brand new post!",
      image: "https://via.placeholder.com/150",
      username: "Buck",
      subreddit_id: 1,
      subreddit_topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertPost: {
        __typename: "Post",
        id: 1,
        title: "This is a new post",
        body: "A brand new post!",
        image: "https://via.placeholder.com/150",
        username: "Buck",
        subreddit_id: 1,
        subreddit_topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
        votes: [
          {
            __typename: "Vote",
            id: -1,
            post_id: 1,
            username: "Buck",
            upvote: true,
          },
        ],
      },
    },
  },
};
