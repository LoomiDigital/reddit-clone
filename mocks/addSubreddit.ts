import { MockedResponse } from "@apollo/client/testing";
import { AddSubredditDocument } from "@d20/generated/graphql";

export const mockAddSubreddit: MockedResponse = {
  request: {
    query: AddSubredditDocument,
    variables: {
      topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertSubreddit: {
        __typename: "Subreddit",
        id: 1,
        post_id: 1,
        topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  },
};
