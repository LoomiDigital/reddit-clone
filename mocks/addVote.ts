import { MockedResponse } from "@apollo/client/testing";
import { AddVoteDocument } from "@d20/generated/graphql";

export const mockAddVote: MockedResponse = {
  request: {
    query: AddVoteDocument,
    variables: {
      post_id: 1,
      username: "Buck",
      upvote: true,
    },
  },
  result: {
    data: {
      addVote: {
        __typename: "Vote",
        post_id: 1,
        upvote: true,
        username: "Buck",
      },
    },
  },
};
