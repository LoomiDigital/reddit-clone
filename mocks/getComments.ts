import { MockedResponse } from "@apollo/client/testing";
import {
  GetCommentsByPostIdDocument,
  GetPostsDocument,
  GetPostsQuery,
  PostConnection,
} from "@d20/generated/graphql";

export const mockCommentsResponse: MockedResponse = {
  request: {
    query: GetCommentsByPostIdDocument,
    variables: {
      post_id: 1,
    },
  },
  result: {
    data: {
      commentsByPostId: [
        {
          id: 1,
          text: "a comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
        {
          id: 2,
          text: "another comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
      ],
    },
  },
};

export const mockPosts: GetPostsQuery = {
  posts: {
    __typename: "PostConnection",
    edges: [
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 2,
          title: "Test Post",
          body: "Test Content",
          username: "Buck",
          image: "http://placehold.it/300x300",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 2,
            },
          ],
          subreddit_id: 2,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    ],
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    },
  },
};

export const mockPostConnection: PostConnection = {
  __typename: "PostConnection",
  edges: [
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 1,
        title: "Test Post",
        body: "Test Content",
        image: "http://placehold.it/300x300",
        username: "Buck",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 1,
          },
        ],
        subreddit_id: 1,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 2,
        title: "Test Post",
        body: "Test Content",
        username: "Buck",
        image: "http://placehold.it/300x300",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 2,
          },
        ],
        subreddit_id: 2,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  ],
  pageInfo: {
    __typename: "PageInfo",
    hasPreviousPage: false,
    hasNextPage: false,
    startCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
  },
};
