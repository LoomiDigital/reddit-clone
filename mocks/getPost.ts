import { MockedResponse } from "@apollo/client/testing";
import {
  GetPostDocument,
  PostAttributesFragment,
} from "@d20/generated/graphql";

export const mockPostResponse: MockedResponse = {
  request: {
    query: GetPostDocument,
    variables: {
      id: 1,
    },
  },
  result: {
    data: {
      getPost: {
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    },
  },
};

export const mockPost: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: false,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};

export const mockPostUpvoted: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: true,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};
