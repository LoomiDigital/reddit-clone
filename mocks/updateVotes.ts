import { MockedResponse } from "@apollo/client/testing";
import { UpdateVoteDocument } from "@d20/generated/graphql";

export const mockUpvote: MockedResponse = {
  request: {
    query: UpdateVoteDocument,
    variables: {
      post_id: 1,
      upvote: true,
      username: "Buck",
    },
  },
  result: {
    data: {
      updateVote: {
        post_id: 1,
        id: 1,
        username: "Buck",
        upvote: true,
      },
    },
  },
};

export const mockDownvote: MockedResponse = {
  request: {
    query: UpdateVoteDocument,
    variables: {
      post_id: 1,
      upvote: false,
      username: "Buck",
    },
  },
  result: {
    data: {
      updateVote: {
        __typename: "Vote",
        post_id: 1,
        id: 1,
        username: "Buck",
        upvote: false,
      },
    },
  },
};
