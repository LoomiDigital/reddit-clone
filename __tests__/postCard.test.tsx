import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SessionProvider } from "next-auth/react";
import { MockedProvider } from "@apollo/client/testing";

import {
  mockPost,
  mockPostResponse,
  mockPostUpvoted,
} from "@d20/mocks/getPost";
import { mockCommentsResponse } from "@d20/mocks/getComments";
import { mockDownvote, mockUpvote } from "@d20/mocks/updateVotes";

import PostCard from "@d20/Components/PostCard";

describe("PostCard component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[mockPostResponse, mockCommentsResponse]}>
          <SessionProvider
            session={{
              expires: "2021-10-10",
              user: {
                name: "Buck",
                expires: "2021-10-10",
                email: "user@test.com",
                address: "123 Fake St",
                image: "https://via.placeholder.com/150",
              },
            }}
          >
            <PostCard post={mockPost} />
          </SessionProvider>
        </MockedProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render a post correctly", async () => {
    render(
      <MockedProvider mocks={[mockPostResponse, mockCommentsResponse]}>
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.placeholder.com/150",
            },
          }}
        >
          <PostCard post={mockPost} />
        </SessionProvider>
      </MockedProvider>
    );

    expect(await screen.findByText("Test Post")).toBeInTheDocument();
  });

  it("should cast an upvote", async () => {
    render(
      <MockedProvider
        mocks={[mockCommentsResponse, mockPostResponse, mockUpvote]}
      >
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.placeholder.com/150",
            },
          }}
        >
          <PostCard post={mockPost} />
        </SessionProvider>
      </MockedProvider>
    );

    const upvoteButton = await screen.findByTitle("upvote");

    fireEvent.click(upvoteButton);

    expect((await screen.findByTitle("votes")).innerHTML).toBe("1");
    expect(upvoteButton.parentElement).toHaveClass("text-red-400");
  });

  it("should cast a downvote", async () => {
    render(
      <MockedProvider
        mocks={[mockCommentsResponse, mockPostResponse, mockDownvote]}
      >
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.pla ceholder.com/150",
            },
          }}
        >
          <PostCard post={mockPostUpvoted} />
        </SessionProvider>
      </MockedProvider>
    );

    const downButton = screen.getByTitle("downvote");

    fireEvent.click(downButton);

    expect((await screen.findByTitle("votes")).innerHTML).toBe("-1");
    expect(downButton.parentElement).toHaveClass("text-blue-400");
  });

  it("should display the correct number of comments", async () => {
    render(
      <MockedProvider mocks={[mockCommentsResponse]}>
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.pla ceholder.com/150",
            },
          }}
        >
          <PostCard post={mockPost} />
        </SessionProvider>
      </MockedProvider>
    );

    expect(await screen.findByText("2 Comments")).toBeInTheDocument();
  });
});
