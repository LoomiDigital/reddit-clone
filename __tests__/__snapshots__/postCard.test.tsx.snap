// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard component should render correctly 1`] = `
<a
  href="/post/1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="flex cursor-pointer rounded-md border border-gray-300 bg-white shadow-sm hover:border hover:border-gray-600"
  >
    <div
      className="flex flex-col items-center justify-start space-y-1 rounded-l-md bg-gray-50 p-4 text-gray-400"
    >
      <svg
        aria-hidden="true"
        className="voteButtons cursor-pointer hover:text-red-400 undefined"
        fill="none"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          upvote
        </title>
        <path
          d="M4.5 10.5L12 3m0 0l7.5 7.5M12 3v18"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
      <p
        className="votes text-xs font-bold text-black"
        title="votes"
      >
        0
      </p>
      <svg
        aria-hidden="true"
        className="voteButtons cursor-pointer hover:text-blue-400 text-blue-400"
        fill="none"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          downvote
        </title>
        <path
          d="M19.5 13.5L12 21m0 0l-7.5-7.5M12 21V3"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
    <div
      className="p-3 pb-1"
    >
      <div
        className="flex items-center space-x-2"
      >
        <div
          className="relative h-10 w-10 overflow-hidden rounded-full border-gray-300 bg-white undefined"
        >
          <img
            alt="Buck"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="https://avatars.dicebear.com/api/open-peeps/test.svg"
            style={
              {
                "bottom": 0,
                "color": "transparent",
                "height": "100%",
                "left": 0,
                "objectFit": undefined,
                "objectPosition": undefined,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
        <p
          className="text-xs text-gray-400"
        >
          <span
            className="font-bold text-black hover:text-blue-400 hover:underline"
            onClick={[Function]}
          >
            r/
            test
          </span>
          <span>
            â€¢ Posted by u/
            Buck
             
            Loading...
          </span>
        </p>
      </div>
      <div
        className="py-4"
      >
        <h2
          className="text-xl font-semibold"
        >
          Test Post
        </h2>
        <p
          className="mt-2 text-sm font-light"
        >
          Test Content
        </p>
      </div>
      <img
        alt=""
        className="w-full"
        src="http://placehold.it/300x300"
      />
      <div
        className="flex space-x-4 text-gray-400 "
      >
        <div
          className="postButtons"
        >
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.625 12a.375.375 0 11-.75 0 .375.375 0 01.75 0zm0 0H8.25m4.125 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zm0 0H12m4.125 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zm0 0h-.375M21 12c0 4.556-4.03 8.25-9 8.25a9.764 9.764 0 01-2.555-.337A5.972 5.972 0 015.41 20.97a5.969 5.969 0 01-.474-.065 4.48 4.48 0 00.978-2.025c.09-.457-.133-.901-.467-1.226C3.93 16.178 3 14.189 3 12c0-4.556 4.03-8.25 9-8.25s9 3.694 9 8.25z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
          <p>
            0
             Comments
          </p>
        </div>
        <div
          className="postButtons"
        >
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 11.25v8.25a1.5 1.5 0 01-1.5 1.5H5.25a1.5 1.5 0 01-1.5-1.5v-8.25M12 4.875A2.625 2.625 0 109.375 7.5H12m0-2.625V7.5m0-2.625A2.625 2.625 0 1114.625 7.5H12m0 0V21m-8.625-9.75h18c.621 0 1.125-.504 1.125-1.125v-1.5c0-.621-.504-1.125-1.125-1.125h-18c-.621 0-1.125.504-1.125 1.125v1.5c0 .621.504 1.125 1.125 1.125z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
          <p
            className="hidden sm:inline"
          >
            Award
          </p>
        </div>
        <div
          className="postButtons"
        >
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.217 10.907a2.25 2.25 0 100 2.186m0-2.186c.18.324.283.696.283 1.093s-.103.77-.283 1.093m0-2.186l9.566-5.314m-9.566 7.5l9.566 5.314m0 0a2.25 2.25 0 103.935 2.186 2.25 2.25 0 00-3.935-2.186zm0-12.814a2.25 2.25 0 103.933-2.185 2.25 2.25 0 00-3.933 2.185z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
          <p
            className="hidden sm:inline"
          >
            Share
          </p>
        </div>
        <div
          className="postButtons"
        >
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.593 3.322c1.1.128 1.907 1.077 1.907 2.185V21L12 17.25 4.5 21V5.507c0-1.108.806-2.057 1.907-2.185a48.507 48.507 0 0111.186 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
          <p
            className="hidden sm:inline"
          >
            Save
          </p>
        </div>
        <div
          className="postButtons"
        >
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</a>
`;
